test/functional/runner/index.js

```diff
@@ -0,0 +1,590 @@
+'use strict';
+const Promise = require('bluebird');
+const path = require('path');
+const fs = require('fs');
+const chai = require('chai');
+const expect = chai.expect;
+const EJSON = require('mongodb-extjson');
+const TestRunnerContext = require('./context').TestRunnerContext;
+const resolveConnectionString = require('./utils').resolveConnectionString;
+
+chai.use(require('chai-subset'));
+chai.use(require('./matcher').default);
+chai.config.includeStack = true;
+chai.config.showDiff = true;
+chai.config.truncateThreshold = 0;
+
+function escape(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+function isPlainObject(value) {
+  return value !== null && typeof value === 'object' && Array.isArray(value) === false;
+}
+
+function translateClientOptions(options) {
+  Object.keys(options).forEach(key => {
+    if (key === 'readConcernLevel') {
+      options.readConcern = { level: options.readConcernLevel };
+      delete options[key];
+    } else if (key === 'autoEncryptOpts') {
+      options.autoEncryption = Object.assign({}, options.autoEncryptOpts);
+
+      if (options.autoEncryptOpts.kmsProviders) {
+        delete options.kmsProviders;
+        options.autoEncryption.kmsProviders = options.autoEncryptOpts.kmsProviders;
+
+        const kmsProviders = options.autoEncryption.kmsProviders;
+        if (kmsProviders.aws) {
+          const awsProvider = kmsProviders.aws;
+          awsProvider.accessKeyId = process.env.AWS_ACCESS_KEY_ID || 'NOT_PROVIDED';
+          awsProvider.secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY || 'NOT_PROVIDED';
+        }
+
+        if (kmsProviders.local && kmsProviders.local.key) {
+          const localKey = kmsProviders.local.key;
+          if (localKey._bsontype && localKey._bsontype === 'Binary' && localKey.sub_type === 0) {
+            // this is read in as BSON Binary subtype 0, extract the Buffer
+            kmsProviders.local.key = kmsProviders.local.key.buffer;
+          }
+        }
+      }
+
+      delete options.autoEncryptOpts;
+    }
+  });
+
+  return options;
+}
+
+function gatherTestSuites(specPath) {
+  return fs
+    .readdirSync(specPath)
+    .filter(x => x.indexOf('.json') !== -1)
+    .map(x =>
+      Object.assign(EJSON.parse(fs.readFileSync(`${specPath}/${x}`), { relaxed: true }), {
```

Comment:
this should leverage `path.join` to ensure windows support.

Reply:
nice catch, fixed.